### PR TITLE
To support multiple clusters on the same edgecore level.

### PR DESCRIPTION
**Pull request by click2cloud-team3**
**What type of PR is this?**

* Added a feature: 
* Feature details: Extended edgecore clusterd to support multiple clusters on the same edgecore level

                           edgecore (clusterd)
                                     /      \
                                    /        \
                                   /          \
                           cluster A           cluster B

**What this PR does / why we need it**:

 * Earlier clusterd module in edgecore was supporting only single k8s cluster on the same edgecore level.
 * This PR resolves the above issue. Now clusterd is configured to support multiple k8s clusters on the same edgecore cluster.
 * Also updated setup guide.

**Which issue(s) this PR fixes**:

Usage: Fixed issue ([#82](https://github.com/CentaurusInfra/fornax/issues/82)).

**Test details:**

* Mission deployment to all clusters, to a given cluster, and to selective clusters works fine with the following environment setup.

`Case 1:`  One cloudcore configured with one edgecore where edgecores configured with multiple k8s control planes.

`Case 2:` One cloudcore configured with multiple edgecores where edgecores configured with multiple k8s control planes.

`Case 3:` Hierarchical setup of cloudcores and  edgecores where each edgecore is configured with multiple k8s control planes.

### CODE REVIEW

File: edge/pkg/clusterd/mission_deployer.go
Comment (pdgetrf) [inline edge/pkg/clusterd/mission_deployer.go:123]: add a return nil statement here and get rid of the "else"

File: pkg/apis/componentconfig/edgecore/v1alpha1/default.go
Comment (pdgetrf) [inline pkg/apis/componentconfig/edgecore/v1alpha1/default.go:291]: SetKubeconfigArray -> SetKubeConfigArray

File: pkg/apis/componentconfig/edgecore/v1alpha1/default.go
Comment (pdgetrf) [inline pkg/apis/componentconfig/edgecore/v1alpha1/default.go:308]: this name is a bit confusing. it could be renamed to "ConfigFileExist" or something like that

**Review by pdgetrf — APPROVED** [review]
LGTM

a few minor comments.

@jshaofuturewei from our team will also review this. once we have both approval and all comments addressed, we can merge this.

File: pkg/apis/componentconfig/edgecore/v1alpha1/types.go
Comment (jshaofuturewei) [inline pkg/apis/componentconfig/edgecore/v1alpha1/types.go:492]: clusters

File: pkg/apis/componentconfig/edgecore/v1alpha1/types.go
Comment (jshaofuturewei) [inline pkg/apis/componentconfig/edgecore/v1alpha1/types.go:493]: Do we need to add json mappings here?

**Review by jshaofuturewei — APPROVED** [review]

File: edge/pkg/clusterd/mission_deployer.go
Comment (click2cloud-team3) [reply edge/pkg/clusterd/mission_deployer.go]: Updated, thanks

File: pkg/apis/componentconfig/edgecore/v1alpha1/types.go
Comment (click2cloud-team3) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go]: Updated, thanks

File: pkg/apis/componentconfig/edgecore/v1alpha1/types.go
Comment (click2cloud-team3) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go]: Added suggested changes, thanks

File: edge/pkg/clusterd/mission_deployer.go
Comment (click2cloud-team3) [reply edge/pkg/clusterd/mission_deployer.go]: Made appropriate changes, thanks

File: docs/setup-guide/cluster_setup.md
Comment (click2cloud-team3) [reply docs/setup-guide/cluster_setup.md]: updated, thanks

File: pkg/apis/componentconfig/edgecore/v1alpha1/default.go
Comment (click2cloud-team3) [reply pkg/apis/componentconfig/edgecore/v1alpha1/default.go]: Updated name of the function, thanks

File: pkg/apis/componentconfig/edgecore/v1alpha1/default.go
Comment (click2cloud-team3) [reply pkg/apis/componentconfig/edgecore/v1alpha1/default.go]: Updated name of the function, thanks

File: edge/pkg/clusterd/mission_deployer.go
Comment (click2cloud-team3) [reply edge/pkg/clusterd/mission_deployer.go:124]: Made appropriate changes as per suggestions, thanks

**Pull request merged**
